kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/filesystem/CSVUtil.java

```diff
@@ -141,10 +156,19 @@ private static Schema inferSchemaInternal(String name, InputStream incoming,
       for (int i = 0; i < header.length; i += 1) {
         if (i < line.length) {
           if (types[i] == null) {
-            types[i] = inferFieldType(line[i]);
-            if (types[i] != null) {
-              // keep track of the value used
-              values[i] = line[i];
+            String fieldName = header[i].trim();
+            if (specifiedTypes != null && specifiedTypes.containsKey(fieldName)) {
+              types[i] = stringToFieldType(specifiedTypes.get(fieldName));
+              if (types[i] != null) {
+                // reconstruct command line arg as the value used
+                values[i] = "--field-type " + fieldName + "=" + specifiedTypes.get(fieldName);
```

Comment:
This doesn't fit the contract of the values array. I see what you're doing here, though. I think the right thing to do is to have a `doc` array so you can set it to "Type inferred from '...'" in one case and "Type set by ..." in the other case.